CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLHarvester.cc

```diff
@@ -148,53 +202,126 @@ void SiPixelLorentzAnglePCLHarvester::dqmEndJob(DQMStore::IBooker& iBooker, DQMS
   MonitorElement* h_drift_depth_adc_slice_ =
       iBooker.book1D("h_drift_depth_adc_slice", "slice of adc histogram", hist_drift_, min_drift_, max_drift_);
 
-  TF1* f1 = new TF1("f1", "[0] + [1]*x", 50., 235.);
-  f1->SetParName(0, "p0");
-  f1->SetParName(1, "p1");
-  f1->SetParameter(0, 0);
-  f1->SetParameter(1, 0.4);
   std::cout << "module"
             << "\t"
             << "layer"
             << "\t"
             << "offset"
             << "\t"
-            << "error"
+            << "e0"
             << "\t"
             << "slope"
             << "\t"
-            << "error"
+            << "e1"
             << "\t"
                "rel.err"
             << "\t"
                "pull"
             << "\t"
+            << "p2"
+            << "\t"
+            << "e2"
+            << "\t"
+            << "p3"
+            << "\t"
+            << "e3"
+            << "\t"
+            << "p4"
+            << "\t"
+            << "e4"
+            << "\t"
+            << "p5"
+            << "\t"
+            << "e5"
+            << "\t"
             << "chi2"
             << "\t"
-            << "prob" << std::endl;
+            << "prob"
+            << "\t"
+            << "newDetId" << std::endl;
 
   SiPixelLorentzAngle* LorentzAngle = new SiPixelLorentzAngle();
 
+  TF1* f1 = new TF1("f1", "[0] + [1]*x + [2]*x*x + [3]*x*x*x + [4]*x*x*x*x + [5]*x*x*x*x*x", 5., 280.);
```

Comment:
using smart pointers is recommended.

Reply:
For this line: https://github.com/CMSTrackerDPG/cmssw/blob/23446a2dce849ae9370e3cc5579e1bf1d08d446d/CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLHarvester.cc#L264

I got an compilation error if I use the smart pointer 

```
no matching function for call to 'TH1::Fit(std::unique_ptr<TF1>&, const char [4])
```

Reply:
I think you can do `f1.get()` in order to get the underlying raw pointer and use it for the fit.